src/archive/zip.rs

```diff
@@ -219,10 +219,16 @@ where
                     options
                 };
 
-                let mut file = fs::File::open(entry.path())?;
+                let mut file = fs::File::open(path)?;
+                let large = file.metadata().map_or(
+                    true,
+                    |metadata| metadata.len() > 0xffffffff, // 4 GB
```

Comment:
Maybe we should just put `true` for every file, `ZIP64` should be very common, what do you think?

From [the documentation](https://docs.rs/zip/latest/zip/write/struct.FileOptions.html#method.large_file) this potentially wastes `20b`, for a thousand files this sums up to `2KB`, which is equivalent to 4 blocks in my filesystem (block size is `512b`), for a thousand-file archive, `2kb` seems irrelevant.

About micro-optimizing this, if we just pick `true`, maybe Rust can do some inlining and remove the branch of dealing with the `false` case.

Reply:
you've convinced me, `File::metadata` seems to require syscalls and the extra 20B might be worth it until we implement multithreaded zip compression